src/Cardano/Wallet/CoinSelection/Fee.hs

```diff
@@ -341,3 +347,96 @@ computeFee (CoinSelection inps outs chgs) =
     -- by definition, impossible; unless we messed up real hard.
     collapse [] _ =
         invariant "outputs are bigger than inputs" (undefined) (const False)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size in bytes, @a@ and @b@ are
+-- constant coefficients.
+data TxSizeLinear = TxSizeLinear Double Double
+    deriving (Eq, Show)
+
+cardanoPolicy :: TxSizeLinear
+cardanoPolicy = TxSizeLinear 155381 43.946
+
+-- | Estimate the fee for a transaction that has @inps@ inputs
+-- and @outps@ outputs as coming from coin selection.
+--
+-- NOTE: The average size of @Attributes AddrAttributes@ and
+--       the transaction attributes @Attributes ()@ are both hard-coded
+estimateCardanoFee
+    :: TxSizeLinear
+    -> CoinSelection
+    -> Fee
+estimateCardanoFee (TxSizeLinear a b) (CoinSelection inps outs _) =
+    Fee $ multioutputCorrection + ceiling (a + b*totalPayload)
+    where
+
+        multioutputCorrection :: Word64
+        multioutputCorrection = case length outs of
+            1 -> 0
+            _ -> 1
+
+        -- The size of TxIn is always the same as it contains the hash and index
+        -- see Cardano.Wallet.Binary (encodeTxIn)
+        sizeOfTxIn :: Int
+        sizeOfTxIn = 42
+
+        sizeOfListLen :: Int
+        sizeOfListLen = 1
+
+        sizeOfListBreak :: Int
+        sizeOfListBreak = 1
+
+        -- The size of [TxIn]
+        sizeOfTxIns :: Int
+        sizeOfTxIns =
+            let n = length inps
+            in sizeOfListLen + n*sizeOfTxIn + (n-1)*sizeOfListBreak + (n-1)*2
```

Comment:
`(n-1)*2` is added artificially here - I cannot explain it at the moment. Just added in order to obtain the same results as with `cardano-sl`